packages/cli/src/ui/components/messages/ToolMessage.tsx

```diff
@@ -62,11 +62,44 @@ export const ToolMessage: React.FC<ToolMessageProps> = ({
     ptyId === activeShellPtyId &&
     shellFocused;
 
+  const [lastUpdateTime, setLastUpdateTime] = React.useState<Date | null>(null);
+  const [userHasFocused, setUserHasFocused] = React.useState(false);
+  const [showFocusHint, setShowFocusHint] = React.useState(false);
+
+  React.useEffect(() => {
+    if (resultDisplay) {
+      setLastUpdateTime(new Date());
+    }
+  }, [resultDisplay]);
+
+  React.useEffect(() => {
+    const interval = setInterval(() => {
```

Comment:
doing this with setInterval seems like a bit of overkill. why can't we use setTimeout instead of polling once every seconds for 5 seconds to see if the timeout has expired.

Reply:
The reason i used polling is because this checks that there has been a 5 second pause, after receiving data, so i don't think a normal setTimeout would work here